end2end.test/src/test/scala/com/expedia/adaptivealerting/pipeline/test/ConstantThresholdBasedE2ETestSpec.scala

```diff
@@ -0,0 +1,118 @@
+  protected var anomalyTopicConsumerConfig = consumerConfig("anomalies-consumer","anomaly")
+
+
+  override def beforeEach() {
+    EmbeddedKafka.CLUSTER.createTopic(INPUT_TOPIC)
+    EmbeddedKafka.CLUSTER.createTopic(INTERMEDIATE_TOPIC)
+    EmbeddedKafka.CLUSTER.createTopic(OUTPUT_TOPIC)
+    IntegrationTestUtils.purgeLocalStreamsState(metricRouterStreamConfig)
+    IntegrationTestUtils.purgeLocalStreamsState(constantThresholdStreamConfig)
+  }
+
+  override def afterEach(): Unit = {
+    EmbeddedKafka.CLUSTER.deleteTopic(INPUT_TOPIC)
+    EmbeddedKafka.CLUSTER.deleteTopic(INTERMEDIATE_TOPIC)
+    EmbeddedKafka.CLUSTER.deleteTopic(OUTPUT_TOPIC)
+  }
+
+  "AA with 'Constant Threshold Anomaly Detection' model" should {
+
+    "consume anomalous metrics from input topic, classify and confirm them as anomalies and " +
+      "send results to output topic" in {
+
+      Given("a set of anomalous metrics and kafka specific configurations")
+      val metrics = generateAnomalousMetrics()
+      val metricRouterSR = metricRouterStreamRunner
+      val constantThresholdSR = constantThresholdDetectorStreamRunner
+
+      When("anomalous metrics are produced in 'input' topic, and kafka-streams topology is started")
+      produceSpansAsync(10.millis, metrics)
+      metricRouterSR.start()
+      constantThresholdSR.start()
+
+      Then("'metric router' should route the metrics to 'constant threshold outlier detector' model's input topic")
+
+      val intermediateRecords: List[KeyValue[String, MetricPoint]] =
+        IntegrationTestUtils.waitUntilMinKeyValueRecordsReceived[String, MetricPoint](
+          configToProps(constantThresholdTopicConsumerConfig.getConfig(STREAM)),
+          INTERMEDIATE_TOPIC, 2, 15000).asScala.toList // get metricPoints from Kafka's output topic
+      intermediateRecords.size shouldEqual 2
+      intermediateRecords.foreach(i => {
+        List("latency", "duration") contains(i.value.metric) shouldBe true
+      })
+
+      Then("'constant threshold outlier detector' should read records from its topic and " +
+        "write those anomalous records to output topic")
+      val consumerPropAnomalyTopic = configToProps(anomalyTopicConsumerConfig.getConfig(STREAM))
+      consumerPropAnomalyTopic.put("JsonPOJOClass", classOf[AnomalyResult])
+      val outputRecords: List[KeyValue[String, AnomalyResult]] =
+        IntegrationTestUtils.waitUntilMinKeyValueRecordsReceived[String, AnomalyResult](
+          consumerPropAnomalyTopic,
+          OUTPUT_TOPIC, 2, 35000).asScala.toList // get metricPoints from Kafka's output topic
+      outputRecords.size shouldEqual 2
+
+      Then("no other intermediate partitions are created after as a result of topology")
+      val adminClient: AdminClient = AdminClient.create(metricRouterStreamConfig)
+      val topicNames: Iterable[String] = adminClient.listTopics.listings().get().asScala
```

Comment:
get can throw a none.get exception, would recommend adding an assertion that the listings exist before you do a get. this would help your test fail gracefully in case the topics aren't created